include/powsybl/math/UndirectedGraph.hxx

```diff
@@ -450,6 +450,14 @@ void UndirectedGraph<V, E>::traverse(unsigned long v, const Traverser& traverser
     }
 }
 
+template <typename V, typename E>
+bool UndirectedGraph<V, E>::vertexExists(unsigned long v) const {
+    if (v >= m_vertices.size()) {
+        throw PowsyblException(stdcxx::format("Invalid vertex %1%", v));
+    }
+    return static_cast<bool>(m_vertices[v]);
+}
```

Comment:
This is not consistent with the java code: In java, if `v` is greater or equal to `m_vertices.size()`, it returns false, it doesn't throw an exception.
As C++ makes lazy evaluation, you can simply copy/paste the java code

Reply:
`v` is an `unsigned long` so the method cannot throw any exception unlike in Java.

Reply:
Agree, but in your first version you would throw an exception if `v` is greater than the vertices size. This was my point. The java code, in that specific case returned false. I see that in the last version you fix it.